From bec1b0ced72dc0d49825d6b69d09296d13fd0af3 Mon Sep 17 00:00:00 2001
From: Andre Przywara <andre.przywara@arm.com>
Date: Tue, 21 Jul 2015 10:58:46 +0100
Subject: [PATCH] avoid redefining PAGE_SIZE

PAGE_SIZE may have been defined by the C libary (musl-libc does that).
So avoid redefining it here unconditionally, instead only use our
definition if none has been provided by the libc.

Signed-off-by: Andre Przywara <andre.przywara@arm.com>
Signed-off-by: Will Deacon <will.deacon@arm.com>
[backport from upstream commit 4095fac878f618ae5e7384a1dc65ee34b6e05217.]
Signed-off-by: Thomas Petazzoni <thomas.petazzoni@free-electrons.com>
---
 include/kvm/kvm.h | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/include/kvm/kvm.h b/include/kvm/kvm.h
index 754e029..37155db 100644
--- a/include/kvm/kvm.h
+++ b/include/kvm/kvm.h
@@ -11,6 +11,7 @@
 #include <time.h>
 #include <signal.h>
 #include <sys/prctl.h>
+#include <limits.h>
 
 #define SIGKVMEXIT		(SIGRTMIN + 0)
 #define SIGKVMPAUSE		(SIGRTMIN + 1)
@@ -19,7 +20,9 @@
 #define HOME_DIR		getenv("HOME")
 #define KVM_BINARY_NAME		"lkvm"
 
+#ifndef PAGE_SIZE
 #define PAGE_SIZE (sysconf(_SC_PAGE_SIZE))
+#endif
 
 #define DEFINE_KVM_EXT(ext)		\
 	.name = #ext,			\
-- 
2.9.4

